pennylane/beta/tapes/qnode.py

```diff
@@ -0,0 +1,541 @@
+              devices that are classically end-to-end differentiable, for example
+              :class:`default.tensor.tf <~.DefaultTensorTF>`. Note that the returned
+              QNode can only be used with the machine-learning framework supported
+              by the device; a separate ``interface`` argument should not be passed.
+
+            * ``"reversible"``: Uses a reversible method for computing the gradient.
+              This method is similar to ``"backprop"``, but trades off increased
+              runtime with significantly lower memory usage. Compared to the
+              parameter-shift rule, the reversible method can be faster or slower,
+              depending on the density and location of parametrized gates in a circuit.
+              Only allowed on (simulator) devices with the "reversible" capability,
+              for example :class:`default.qubit <~.DefaultQubit>`.
+
+            * ``"device"``: Queries the device directly for the gradient.
+              Only allowed on devices that provide their own gradient rules.
+
+            * ``"parameter-shift"``: Use the analytic parameter-shift
+              rule where possible, with finite-difference as a fallback.
+
+            * ``"finite-diff"``: Uses numerical finite-differences for all parameters.
+
+    Keyword Args:
+        h=1e-7 (float): Step size for the finite difference method.
+        order=1 (int): The order of the finite difference method to use. ``1`` corresponds
+            to forward finite differences, ``2`` to centered finite differences.
+
+    **Example**
+
+    >>> from pennylane.beta.queuing import expval, var, sample, probs
+    >>> from pennylane.beta.tapes import QNode
+    >>> def circuit(x):
+    >>>     qml.RX(x, wires=0)
+    >>>     return expval(qml.PauliZ(0))
+    >>> dev = qml.device("default.qubit", wires=1)
+    >>> qnode = QNode(circuit, dev)
+    """
+
+    # pylint:disable=too-many-instance-attributes
+
+    def __init__(self, func, device, interface="autograd", diff_method="best", **diff_options):
+
+        if interface is not None and interface not in self.INTERFACE_MAP:
+            raise QuantumFunctionError(
+                f"Unknown interface {interface}. Interface must be "
+                f"one of {self.INTERFACE_MAP.values()}."
+            )
+
+        if not isinstance(device, Device):
+            raise QuantumFunctionError("Invalid device. Device must be a valid PennyLane device.")
+
+        self.func = func
+        self.device = device
+        self.qtape = None
+
+        self._tape, self.interface, self.diff_method = self._get_tape(
+            device, interface, diff_method
+        )
+        self.diff_options = diff_options or {}
+        self.diff_options["method"] = self.diff_method
+
+        self.dtype = np.float64
+        self.max_expansion = 2
+
+    @staticmethod
+    def _get_tape(device, interface, diff_method="best"):

```

Comment:
These are static methods, since they do not depend on the internal state. At the same time, I think it makes sense to have them as staticmethods rather than functions --- this is better for organization, and makes them more user/developer facing.

Reply:
Although, the underscore means that it won't be seen by users in the docs.

Reply:
oh yeah 😢 Should we change this?

Reply:
done